WordPress/src/main/java/org/wordpress/android/ui/sitecreation/previews/SitePreviewViewModel.kt

```diff
@@ -77,7 +83,8 @@ class SitePreviewViewModel @Inject constructor(
         require(siteCreationState.result is Created)
         siteDesign = siteCreationState.siteDesign
         result = siteCreationState.result
-        urlWithoutScheme = result.site.url
+        isFree = requireNotNull(siteCreationState.domain).isFree
```

Comment:
I guess based on the require here that this isn't ever expected to be `null` here?

Reply:
That's true, the only reason it's null is because of how we modelled the `SiteCreationState` 😄 . Ideally we'd have found a better, immutable way to do it instead of simply storing the value in a nullable field when advancing to the next step of the flow, imho 😄